application.py

```diff
@@ -613,10 +617,14 @@ def create_graph(output_cellset, includeisolated=False, graphname="cellgraph"):
     """
     global e
     # Graph using EM2 or use metadata coordinates
-    if application.config["GRAPH_EM2"]:
+    if application.config["GRAPH_METHOD"] == "em2":
         graph, cellidlist = create_graph_em2(output_cellset, includeisolated=False, graphname="cellgraph")
-    else:
+    elif application.config["GRAPH_METHOD"] == "umap":
+        graph, cellidlist = create_graph_umap(output_cellset)
+    elif application.config["GRAPH_METHOD"] == "tsne":
         graph, cellidlist = create_graph_tsne(output_cellset)
+    else:
+        raise ValueError("Invalid graph method :{}".format(application.config["GRAPH_METHOD"]))
```

Comment:
is this py36+?  if so, you can use f-style strings!

Reply:
Wow! I knew javascript had these, I didn't realize python had gotten onboard

Reply:
Looks like I'm on 3.5